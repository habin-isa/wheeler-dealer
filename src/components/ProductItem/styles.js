import styled from 'styled-components';

export const Wrapper = styled.div`
  align-items: center;
  padding: 30px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const ImgContainer = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  z-index: 9999;
`;

export const Heart = styled.img`
  position: absolute;
  width: 40px;
  padding: 20px;
  z-index: 9999;
  cursor: pointer;
`;

export const ProductImg = styled.img`
  height: 400px;
  width: auto;
`;

export const SoldCaption = styled.div`
  position: absolute;
  font-size: 30px;
  left: 45%;
  top: 40%;
`;

export const ProductInfo = styled.div`
  text-align: left;
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 10px;
`;