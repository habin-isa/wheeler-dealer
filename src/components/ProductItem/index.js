import React, { useState } from 'react';
import * as S from './styles';
import emptyLike from '../../assets/empty-like.svg';
import fullLike from '../../assets/full-like.svg';

const ProductItem = ({ product, handleLikeCount, handleLikeTitle }) => {
  const [itemLiked, setItemLiked] = useState(false);
  const productSold = product.sold;

  const handleLikeClick = () => {
    setItemLiked(!itemLiked);
    if (itemLiked) {
      handleLikeCount(-1);
    } else {
      handleLikeCount(1);
    }
  };

  return (
    <S.Wrapper>
      <S.ImgContainer>
        {
          itemLiked ? (
            <S.Heart src={fullLike} onClick={() => {
              handleLikeClick();
              handleLikeTitle('');
            }} />
          ) : (
            <S.Heart src={emptyLike} onClick={() => {
              handleLikeClick();
              handleLikeTitle(product.title);
            }} />
          )
        }
          <S.ProductImg style={{ opacity: (productSold ? '0.2' : '1') }} src={product.img} alt={product.title} />
          {
            productSold ? (
              <S.SoldCaption><b>SOLD</b></S.SoldCaption>
            ) : ''
          }
      </S.ImgContainer>
      <S.ProductInfo>
        <div><b>{product.title}</b></div>
        <div>Brand: {product.brand}</div>
        <div>Size: {product.size}</div>
        <div>Price: <b>Â£{product.price}</b></div>
      </S.ProductInfo>
    </S.Wrapper>
  )

};

export default ProductItem;