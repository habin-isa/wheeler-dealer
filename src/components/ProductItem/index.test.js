
import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import ProductItem from '.';

describe('Product Item', () => {

  const props = {
    id: '2',
    title: 'Ralph Lauren Tee',
    brand: 'Ralph Lauren',
    size: 'XL',
    price: 'Â£666',
    img: 'https://picsum.photos/500/?image=0',
    product: {
      sold: true
    }
  };

  it('should render', () => {
    const { container } = render(<ProductItem {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with correct product data', () => {
    shallow(<ProductItem product={props} />);
  });

});