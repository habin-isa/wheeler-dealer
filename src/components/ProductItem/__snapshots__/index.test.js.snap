// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Item should render 1`] = `
<div
  class="sc-AxjAm jTjmSu"
>
  <div
    class="sc-AxirZ eWRWoQ"
  >
    <img
      class="sc-AxiKw jkbSeY"
      src="empty-like.svg"
    />
    <img
      class="sc-AxhCb dxDRS"
      style="opacity: 0.2;"
    />
    <div
      class="sc-AxhUy wEiDW"
    >
      <b>
        SOLD
      </b>
    </div>
  </div>
  <div
    class="sc-AxgMl kpvDqU"
  >
    <div>
      <b />
    </div>
    <div>
      Brand: 
    </div>
    <div>
      Size: 
    </div>
    <div>
      Price: 
      <b>
        Â£
      </b>
    </div>
  </div>
</div>
`;
